core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -82,7 +87,21 @@
  * SomeIntegrationTest will be instantiated, lifecycle methods (before/after) will be run, and "someTest" will be invoked.
  *
  */
-public class ClusterTestExtensions implements TestTemplateInvocationContextProvider {
+public class ClusterTestExtensions implements TestTemplateInvocationContextProvider, BeforeEachCallback, AfterEachCallback {
+    private static final String METRICS_METER_TICK_THREAD_PREFIX = "metrics-meter-tick-thread";
+    private static final String SCALA_THREAD_PREFIX = "scala-";
+    private static final String FORK_JOIN_POOL_THREAD_PREFIX = "ForkJoinPool";
+    private static final String JUNIT_THREAD_PREFIX = "junit-";
+    private static final String ATTACH_LISTENER_THREAD_PREFIX = "Attach Listener";
+    private static final String PROCESS_REAPER_THREAD_PREFIX = "process reaper";
+    private static final String RMI_THREAD_PREFIX = "RMI";
+    private static final Set<String> SKIPPED_THREAD_PREFIX = Collections.unmodifiableSet(Stream.of(
+            METRICS_METER_TICK_THREAD_PREFIX, SCALA_THREAD_PREFIX, FORK_JOIN_POOL_THREAD_PREFIX, JUNIT_THREAD_PREFIX,
+            ATTACH_LISTENER_THREAD_PREFIX, PROCESS_REAPER_THREAD_PREFIX, RMI_THREAD_PREFIX)
+            .collect(Collectors.toSet()));
+
+    private DetectThreadLeak detectThreadLeak;
```

Comment:
Instance fields are an anti-pattern in junit extensions because they don't permit the extension to be used in concurrent scenarios. You should use the ExtensionContext to store this object per-test.